[Site Isolation][iOS] http/tests/site-isolation/draw-after-cross-origin-navigation-between-two-remote-frames.html is failing

szewai opened the pull request:
#### c9a2af523d0d33a839871df19690e9bb78aa819a
<pre>
[Site Isolation][iOS] http/tests/site-isolation/draw-after-cross-origin-navigation-between-two-remote-frames.html is failing
<a href="https://bugs.webkit.org/show_bug.cgi?id=275065">https://bugs.webkit.org/show_bug.cgi?id=275065</a>
<a href="https://rdar.apple.com/129169433">rdar://129169433</a>

Reviewed by Alex Christensen and Brady Eidson.

In the test, the iframe is first loaded in the mainframe process and switched to a subframe process on navigation. Then
the iframe is switched to another subframe process on second navigation. In the second navigation, the frame&apos;s size is
not propogated to the new subframe process, so the iframe does not render correctly. To fix that, we make UI process
record frame size and send it to new hosting web process when frame load is commited. On web process side, it needs to
update both exposedContentRect and unobscuredContentRect of the FrameView for rendering.

* Source/WebKit/UIProcess/WebFrameProxy.cpp:
(WebKit::WebFrameProxy::commitProvisionalFrame):
* Source/WebKit/UIProcess/WebFrameProxy.h:
(WebKit::WebFrameProxy::setRemoteFrameSize):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::updateRemoteFrameSize):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::updateFrameSize):

Canonical link: <a href="https://commits.webkit.org/279777@main">https://commits.webkit.org/279777@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/309dd98166d11259b24a29317074a7ea4f53627f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/54513 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/33930 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/7081 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/57792 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/59/builds/5246 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/41466 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/5243 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/44147 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/59/builds/5246 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/56608 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32068 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/47210 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25275 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/28868 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/3387 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/50655 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/4752 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/59383 "Built successfully") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/29746 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/4906 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/51570 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/30894 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/47297 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/50940 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/31873 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8066 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/30684 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/UIProcess/WebFrameProxy.h:
```diff
@@ -192,6 +192,7 @@ class WebFrameProxy : public API::ObjectImpl<API::Object::Type::Frame>, public C
     bool hasPendingBackForwardItem() { return m_hasPendingBackForwardItem; }
 
     WebCore::LayerHostingContextIdentifier layerHostingContextIdentifier() const { return m_layerHostingContextIdentifier; }
+    void setFrameSize(WebCore::IntSize size) { m_frameSize = size; }
```
I think this would be more clear if it were like this:

void setRemoteFrameSize(WebCore::IntSize size) { m_remoteFrameSize = size; }

achristensen07 commented on Source/WebKit/WebProcess/WebPage/WebPage.cpp:
```diff
@@ -3347,10 +3347,15 @@ void WebPage::updateFrameSize(WebCore::FrameIdentifier frameID, WebCore::IntSize
     frameView->resize(newSize);
 #if PLATFORM(IOS_FAMILY)
-    // FIXME: This should be removed after rdar://122429810 is fixed.
+    // FIXME: This ensures cross-iframe render correctly;
```
cross-site iframes
cross-iframe doesn't mean anything.

achristensen07 commented on Source/WebKit/WebProcess/WebPage/WebPage.cpp:
```diff
@@ -3347,10 +3347,18 @@ void WebPage::updateFrameSize(WebCore::FrameIdentifier frameID, WebCore::IntSize
+    if (frameView->size() == newSize)
+        return;
+
+    if (frameView->size() == newSize)
```
This is now duplicated.

szewai committed:
[Site Isolation][iOS] http/tests/site-isolation/draw-after-cross-origin-navigation-between-two-remote-frames.html is failing
https://bugs.webkit.org/show_bug.cgi?id=275065
rdar://129169433

Reviewed by Alex Christensen and Brady Eidson.

In the test, the iframe is first loaded in the mainframe process and switched to a subframe process on navigation. Then
the iframe is switched to another subframe process on second navigation. In the second navigation, the frame's size is
not propogated to the new subframe process, so the iframe does not render correctly. To fix that, we make UI process
record frame size and send it to new hosting web process when frame load is commited. On web process side, it needs to
update both exposedContentRect and unobscuredContentRect of the FrameView for rendering.

* Source/WebKit/UIProcess/WebFrameProxy.cpp:
(WebKit::WebFrameProxy::commitProvisionalFrame):
* Source/WebKit/UIProcess/WebFrameProxy.h:
(WebKit::WebFrameProxy::setRemoteFrameSize):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::updateRemoteFrameSize):
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::updateFrameSize):

Canonical link: https://commits.webkit.org/279777@main
